specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/queue.json

```diff
@@ -0,0 +1,591 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- Creates a queue with the given queue name successfully.",
+            "schema": {
+              "$ref": "#/definitions/StorageQueue"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "QueueService"
+        ],
+        "operationId": "GetQueue",
+        "x-ms-examples": {
+          "QueueOperationGet": {
+            "$ref": "./examples/QueueOperationGet.json"
+          }
+        },
+        "description": "Gets the queue with the specified queue name, under the specified account if it exists.",
+        "parameters": [
+          {
+            "$ref": "./storage.json#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "./storage.json#/parameters/StorageAccountName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/QueueName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- returned table with the specified table name successfully.",
+            "schema": {
+              "$ref": "#/definitions/StorageQueue"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "QueueService"
+        ],
+        "operationId": "DeleteQueue",
+        "x-ms-examples": {
+          "QueueOperationDelete": {
+            "$ref": "./examples/QueueOperationDelete.json"
+          }
+        },
+        "description": "Deletes the queue with the specified queue name, under the specified account if it exists.",
+        "parameters": [
+          {
+            "$ref": "./storage.json#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "./storage.json#/parameters/StorageAccountName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/QueueName"
+          }
+        ],
+        "responses": {
```

Comment:
>responses [](start = 9, length = 9)

As per [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#delete-resource](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#delete-resource), you should return 200 in cases where queue existed and was deleted successfully and return 204 when the queue doesn't exist.

Reply:
Hi, Thanks for pointing to the document. This API has already been rolledout. It returns 404 if the resource tobe deleted does not exist and returns 204 when it exists. In the interest of time and JEDI urgency. We can update the response code in later version

Reply:
Is this consistent with the behavior of other APIs in Storage RP?

Reply:
As agreed, will be taken care of in another PR. As server-side is already shipped with this